src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBEnumerableBuilder.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal class DocumentDBEnumerableBuilder<T> : IAsyncConverter<DocumentDBAttribute, IEnumerable<T>> where T : class
+    {
+        private DocumentDBConfiguration _config;
+
+        public DocumentDBEnumerableBuilder(DocumentDBConfiguration config)
+        {
+            _config = config;
+        }
+
+        public async Task<IEnumerable<T>> ConvertAsync(DocumentDBAttribute attribute, CancellationToken cancellationToken)
+        {
+            DocumentDBContext context = _config.CreateContext(attribute);
+
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(context.ResolvedAttribute.DatabaseName, context.ResolvedAttribute.CollectionName);
+
+            List<Document> finalResults = new List<Document>();
+
+            string continuation = null;
+
+            SqlQuerySpec sqlSpec = new SqlQuerySpec
+            {
+                QueryText = context.ResolvedAttribute.SqlQuery,
+                Parameters = context.ResolvedAttribute.SqlQueryParameters ?? new SqlParameterCollection()
+            };
+
+            do
+            {
+                DocumentQueryResponse response = await DocumentDBUtility.RetryAsync(
+                    () => context.Service.ExecuteNextAsync(collectionUri, sqlSpec, continuation),
+                    context.MaxThrottleRetries, codesToIgnore: HttpStatusCode.NotFound);
+
+                if (response == null)
+                {
+                    return null;
```

Comment:
It turns out that 'null' was only used if the collection was not found. I looked at how Table works, and it just rethrows the exception, which I think is more discoverable. 'Null' doesn't really help there. So I've changed this to rethrow instead of using null.